### ISSUE TITLE
Visual Studio 2017: /std:c++17 produces error C2039: 'unexpected' is not a member of 'std'

### ISSUE THREAD

**Issue by anatoly-spb** [ReactiveX/RxCpp#462]
As you know std::unexpected has been deprecated and since C++17 has been removed from the standard C++ library:  (here is details https://en.cppreference.com/w/cpp/error/unexpected) So if we try to build project with /std:c++17 it finished with:
```
include\rxcpp\rx-util.hpp(686): error C2039: 'unexpected': is not a member of 'std'
```

**Comment by kirkshoop** [comment]
Hi! what version of rxcpp is this? This issue was fixed by PR #389.

**Comment by anatoly-spb** [comment]
@kirkshoop I have used rxcpp via conan, there is 4.0.0 (rxcpp/4.0.0@bincrafters/stable). Thank you for the point.
